src/poller.rs

```diff
@@ -1,38 +1,91 @@
-use std::path::Path;
+use std::fs;
+use std::path::{Path, PathBuf};
 use std::process;
 use std::time::Instant;
 
 use tokio::time;
 use tracing::{error, info};
 
-use crate::config::Config;
+use crate::config::{Config, WatchConfig};
 use crate::log::Operation;
 use crate::snapshots;
 
+/// Checks the provided `child_path` is a directory.
+/// If either `includes` or `excludes` are set,
+/// checks whether the path is included/excluded respectively.
+fn is_valid_directory(base_path: &Path, child_path: &PathBuf, value: &WatchConfig) -> bool {
+    if !child_path.is_dir() {
+        return false;
+    }
+
+    let includes = &value.include;
+    let excludes = &value.exclude;
+
+    if includes.len() > 0 {
```

Comment:
It looks like the semantics are currently:

1. Include everything in dir
2. Also include everything from \1 that's also  in `include` list
3. Exclude anything in the exclude list.

It seems like you want to swap 2 & 3:

1. Include everything in dir
2. Exclude anything in the exclude list.
3. Include everything from \1 that's also  in `include` list, even if it were excluded in 2

Thoughts?

Reply:
The semantics currently are more like:

- If `include` is set, only include dirs in the include list
- Else if `exclude` is set, only include dirs NOT in the exclude list
- Else include everything in dir

Each one is mutually exclusive. Perhaps going your 2nd route and having `include` override `exclude` would allow for more control so happy to alter it to go that route.

Reply:
sounds good